File: C#JAVA_Client_Server/ClientServcerC#Protocols/LongExample/Domain/src/main/java/entities/domain/validator/SoferValidator.java
package entities.domain.validator;


import entities.domain.Sofer;
import entities.domain.validator.interfaceAndExceptions.ValidationException;
import entities.domain.validator.interfaceAndExceptions.Validator;

public class SoferValidator implements Validator<Sofer> {
    @Override
    public void validate(Sofer entity) throws ValidationException {
        if(entity.getNumeSofer().length()==0 || entity.getIdEchipa()==null || entity.getIdMasina()==null){
            throw new ValidationException("entities.domain.Sofer INVALID!");
        }
    }
}

File: C#JAVA_Client_Server/ClientServcerC#Protocols/Solution1/Domain/Validators/SoferValidator.cs
using System;
using Domain.validators.interfaceAndException;

namespace Domain.validators
{
    public class SoferValidator : IValidator<Sofer>
    {
        public void validate(Sofer entity)
        {
            if (entity.NumeSofer.Length == 0 || entity.IdEchipa.Equals(null) || entity.IdMasina.Equals(null))
            {
                throw new Exception("Sofer INVALID!");
            }
        }
    }
}